### ISSUE TITLE
Long words in title problematic in content cards

### ISSUE THREAD

**Issue by rtibbles** [learningequality/kolibri#4419]
### Observed behavior
@christianmemije reports:
Long card title get cut off inside cards, fonts popping in making text shift

### Expected behavior
Titles always get displayed in a length sensitive way.
Fonts should not pop in and make text move!

### User-facing consequences
Can't read titles.
Jerky visual experience.

### Steps to reproduce
Look at a content card with a long title.

### Context
* v0.11.0-beta1

**Comment by indirectlylit** [comment]
I'm not reproducing in either Learner or Coach

**Comment by christianmemije** [comment]
This happens for card with long words. We would need to add ` word-break: break-word;`
![localhost_8000_learn_](https://user-images.githubusercontent.com/7193975/47034550-87af9f00-d12c-11e8-9091-97684c018c8e.png)

**Comment by christianmemije** [comment]
Overflowing text fixed in #4439

**Comment by indirectlylit** [comment]
As [indicated here](https://github.com/learningequality/kolibri/issues/5468#issuecomment-491455064), the previous solution caused real words to break at arbitrary points.

One strategy might be to have the `TruncatedText` component gracefully handle overflow, e.g. like:

> ![image](https://user-images.githubusercontent.com/2367265/57562711-00d14680-734a-11e9-92df-0c96dc66c92a.png)

@rtibbles do you see the font movement again after #5479 was merged?

**Comment by rtibbles** [comment]
If anyone else has time to confirm this is still fixed, I would appreciate it.

**Comment by radinamatic** [comment]
On `0.12.y` dev branch I see *`...`* substituting the whole long title of the content card, on both FF & Chrome on Ubuntu 18.04. Not liking the tooltip strategy, it forces the user to hover to figure out what is inside.

![LEDev1905 (start)  Running  - Oracle VM VirtualBox_033](https://user-images.githubusercontent.com/1457929/57986311-a20b7b80-7a73-11e9-9681-947e60d2133d.png)

Words in long titles are still truncated arbitrarily in FF:

![LEDev1905 (start)  Running  - Oracle VM VirtualBox_034](https://user-images.githubusercontent.com/1457929/57986344-03cbe580-7a74-11e9-8305-63f435e068fc.png)

**Comment by indirectlylit** [comment]
>  I see `...` substituting the whole long title of the content card, on both FF & Chrome on Ubuntu 18.04. Not liking the tooltip strategy, it forces the user to hover to figure out what is inside.

Yes, this is the expected behavior at present. In #4439 we tried to address it by breaking at arbitrary points in words, but that led to #5468 which was much more common than this issue, which requires somewhat contrived examples to reproduce.

The main thing I want to make sure of for 0.12.4 is that we don't have this behavior:

> Fonts should not pop in and make text move!

and it sounds like we don't.

Adding the 'needs design' label to hash out possible solutions, e.g. the [one described above](https://github.com/learningequality/kolibri/issues/4419#issuecomment-491464431) (cc @khangmach)

**Comment by indirectlylit** [comment]
Aaactually I'm confused because in your screenshot, it's still breaking across words! This seems to indicate that #5468 was not fixed.

![image](https://user-images.githubusercontent.com/2367265/57994749-7d170880-7a73-11e9-8d3a-6aa751245d80.png)

**Comment by khangmach** [comment]
>Fonts should not pop in and make text move! 

I'm not understanding what this means. Do you mean fonts being loaded which change the length of the title?

>Not liking the tooltip strategy, it forces the user to hover to figure out what is inside.

Without re-designing the entire content card component, having a tooltip sounds like the only reasonable behavior. This comment kind of implies that the user cannot discern what a certain word is without the entire word/string showing. For example, if the word "subtraction" is cut off to "subtrac...", you can reasonably predict what the word is going to be and mentally form the title. For incredibly long nonsense strings/non-western translated words we do need the tooltip shortcut. 

My thoughts:
- No white gradient. This is a weird design choice that I don't see great benefits over an ellipses (...)
- White gradient is incredibly hard to see. And we would still need a tooltip for it, unless we worsen the user experience via introducing horizontal scrolling. We'd also have to do this everywhere else in Kolibri, which is detrimental
- Have Kolibri detect whether a word will cause overflow. If the word causes overflow, move it to a new line
- If on the new line, the word causes overflow, add an ellipsis (...) to cut off the word at the max width of the text area.

![Slice](https://user-images.githubusercontent.com/25045166/58040932-a4a9b780-7aeb-11e9-9332-d95d3f82aa4d.png)


Instead of eliminating tooltips and reinventing the wheel here, we should be encouraging better naming practice and be able to have a consistent shortcut behavior in the Kolibri app to help users discern incredibly long words.

**Comment by indirectlylit** [comment]
There are two scenarios:

1. There are so many words that the title would take more than 3 lines. In this case, we show the ellipsis and things work fine. There is no proposal to change that.
2. There is a single long word which is longer than the width of the card. That's the scenario this issue is dealing with. I've updated the issue title to reflect that. 

Unfortunately I don't know of any straightforward way of adding an ellipsis for this scenario. The gradient was the only thing I could think of but it would be helpful to get some design input

**Comment by indirectlylit** [comment]
closing in favor of https://github.com/learningequality/kolibri/issues/5468
